### DISCUSSION TITLE
Config ignores multiple font variation rules

### DISCUSSION THREAD

**Discussion by romashamin** [ghostty-org/ghostty#3473] [Feature Requests, Ideas]
It seems that if I include multiple `font-variation` rules, the terminal only applies the last one.

In the screenshot, the width is set to 112.5% and the weight to 200. However, the result shows the default font width (75% instead of the expected 112.5%) while the weight is correctly applied.

<img width="401" alt="Image" src="https://github.com/user-attachments/assets/ebaf22fe-15db-4ad5-b4c4-5c3bd91d986f" />

Macos.

**Comment by den-is** [comment]
I confirm on MacOS variation `wght` did not work for me too.
I was able to set weight variation using  `font-style = SemiBold` for `SauceCodePro Nerd Font Mono` (installed on my OS)

**Reply by 00-kat** [reply]
Your issue is likely because `font-variation` only works for variable fonts.  As far as I know, no patched Nerd Font is variable.

**Reply by yaroslav** [reply]
> Your issue is likely because font-variation only works for variable fonts. 

Martian Mono, as mentioned by the OP, is a variable font.

**Reply by 00-kat** [reply]
I don't think `SauceCodePro Nerd Font Mono` is; I was responding specifically to @den-is if that was not clear.

**Comment by mitchellh** [comment]
I'll check this for confirmation soon. This should work fine.

**Comment by ownia** [comment]
Hi there, is there any progress on this issue?

**Comment by starptr** [comment]
The `font-variation = wdth=50` doesn't seem to do anything for me either for Inconsolata in Macos. Perhaps `wdth` doesn't apply correctly and this is making it seem like the first `font-variation` rule is ignored?

**Reply by 00-kat** [reply]
Inconsolata does not appear to be a variable font (but I may be wrong), so `font-variation` wouldn't work.

> Perhaps `wdth` doesn't apply correctly and this is making it seem like the first `font-variation` rule is ignored?

Did a second `font-variation` rule work for you?

**Comment by Arcadi4** [comment]
I am using [GitHub Monaspace](https://monaspace.githubnext.com) and ran into the same issue. Ghostty respects single variable axis settings, but rejects multiple.

Setting more than one axis in the same field will lead to ghostty complaining about invalid config:

```plaintext
font-variation = wdth=100,wght=250
```
<img width="480" height="302" alt="image" src="https://github.com/user-attachments/assets/f296a685-8a86-4a58-a010-fd6b0206d38f" />

By repeating the field, ghostty will just follow the last one:

```
font-variation = wdth=100
font-variation = wght=250 <- Only this one applies
```

Any known workarounds for this issue?
